src/reducers/profile-view.js

```diff
@@ -526,6 +560,33 @@ export const getRightClickedThreadIndex = (state: State) =>
 export const getSelection = (state: State) =>
   getProfileViewOptions(state).selection;
 
+/**
+ * Tracks
+ */
+export const getGlobalTracks = (state: State) =>
+  getProfileView(state).globalTracks;
+export const getGlobalTrackAndIndexByPid = (state: State, pid: Pid) => {
+  const globalTracks = getGlobalTracks(state);
+  const globalTrackIndex = globalTracks.findIndex(
+    track => track.type === 'process' && track.pid === pid
+  );
+  if (globalTrackIndex === -1) {
+    throw new Error('Unable to find the track index for the given pid.');
+  }
+  const globalTrack = globalTracks[globalTrackIndex];
+  if (globalTrack.type !== 'process') {
+    throw new Error('The globalTrack must be a process type.');
+  }
+  return { globalTrackIndex, globalTrack };
```

Comment:
I think we should use `createSelector` because here this returns a new object for each call: not good for pure components !

You can make one of the input selector return the pid, see for an example: https://github.com/devtools-html/perf.html/blob/7a8d8d0dd57a284e17526944b14dd444cbd00256/src/reducers/icons.js#L43-L50

Reply:
This is only true if you actually intend to use it for a React Component later.

Reply:
I added a warning in a comment a selector, but right now it's only actions.